modules/transaction-payment/src/lib.rs

```diff
@@ -807,18 +805,15 @@ where
 				let fee = Self::check_native_is_not_enough(who, fee, reason).map_or_else(|| fee, |amount| amount);
 				let custom_fee_surplus = T::CustomFeeSurplus::get().mul_ceil(fee);
 				let custom_fee_amount = fee.saturating_add(custom_fee_surplus);
-				T::DEX::swap_with_specific_path(
+				T::Swap::swap(
```

Comment:
we still want people to specify path here, so for example it could have better slippage compare to registered path

Reply:
If so, either introduce DEX back or IMO we should add a method which support swap path to `Swap` interface.

Reply:
yeah let's go with second option